arch/xtensa/core/xtensa_vectors.S

```diff
@@ -1161,7 +1161,11 @@ _xt_medint3_exit:
 
 #endif  /* Level 3 */
 
-#if XCHAL_EXCM_LEVEL >= 4
+/* For some reason, the HAL provided by the ESP32 port of FreeRTOS, that
+ * Zephyr uses, defines XCHAL_EXCM_LEVEL to 3.  That essentially enables
+ * the other _Level4Vector routine, that doesn't work on ESP32.
+ */
+#if defined(CONFIG_SOC_ESP32) || (XCHAL_EXCM_LEVEL >= 4)
```

Comment:
can we added a FIXME here and open an issue to track this, maybe there is a better way to fix this later.